src/core/thread/link_metrics.hpp

```diff
@@ -264,6 +264,18 @@ class LinkMetrics : public InstanceLocator, private NonCopyable
      */
     void ProcessEnhAckIeData(const uint8_t *aData, uint8_t aLength, const Neighbor &aNeighbor);
 
+    /**
+     * This method appends MLE Link Metrics Query TLV to a given message.
+     *
+     * @param[in] aMessage              The message to append to.
+     * @param[in] aLinkMetricsContext   A pointer to the link metrics `Context`.
+     *
+     * @retval kErrorNone     Successfully appended the TLV to the message.
+     * @retval kErrorNoBufs   Insufficient buffers available to append the TLV.
+     *
+     */
+    Error AppendLinkMetricsQueryTlv(Message &aMessage, const void *aLinkMetricsContext);
```

Comment:
Should we make `aLinkMetricsContext` a `Context*` instead of `void*`?

Reply:
We can consider it, but then we may need to make `Context` a public definition. May make sense to change this to so we have this in `Mle` only when `LINK_METRCIS` is enabled.

Reply:
Changed this fully in new push. 
- Renamed it to `QueryInfo` instead of `Context`
- `Mle::SendDataRequest()` now has flavor which gets `LinkMetrics::QueryInfo` (only when `LINK_METRICS` is enabled).